src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -347,6 +358,115 @@ def _retrieve_pre_processed_docs(tool: str,
     return docs
 
 
+def _convert_to_iso8601(date_string: str) -> str:
+    # Convert a date string to ISO 8601 format
+    formats_to_try = ["%Y/%m/%d", "%Y-%m-%d"]  # Add more formats as needed
+
+    for date_format in formats_to_try:
+        try:
+            parsed_date = datetime.strptime(date_string, date_format)
+            iso8601_date = parsed_date.date().isoformat()
+            return iso8601_date
+        except ValueError:
+            continue
+
+    raise ValueError("Unrecognized date format")
+
+
+def _convert_values_to_type(docs: list[dict], key: str, col_type: ColumnType):
+    # Convert the values of a column to the specified type
+    values = []
+    for doc in docs:
+        try:
+            value = doc[key]
+            value = None if value in NONE_STR else value
+            if value is None:
+                doc[key] = value
+                continue
+
+            if col_type == ColumnType.INT:
+                doc[key] = int(value)
+            elif col_type == ColumnType.FLOAT:
+                doc[key] = float(value)
+            elif col_type == ColumnType.STRING:
+                doc[key] = str(value)
+            elif col_type == ColumnType.DATE:
+                doc[key] = _convert_to_iso8601(value)
+            elif col_type == ColumnType.ENUM:
+                pass
+            else:
+                raise ValueError(f'casting not implemented for {col_type}')
+
+            values.append(doc[key])
+        except KeyError as e:
+            raise ValueError(f'Unable to find key: {key} in {doc}') from e
+        except ValueError as e:
+            raise ValueError(f'Unable to convert value: {key} from {doc} to type: {col_type}') from e
+
+    return values
+
+
+def process_columnar_meta(
+        docs: list[dict],
+        kbase_collection: str,
+        load_ver: str,
+        allow_missing_metadata: bool = True,
+):
+    """
+    Process the columnar metadata for the genome attributes.
+
+    :param docs: the list of documents
+    :param kbase_collection: the KBase collection name
+    :param load_ver: the load version
+    :param allow_missing_metadata: whether to allow missing metadata
+    """
+
+    spec = load_spec(names.GENOME_ATTRIBS_PRODUCT_ID, kbase_collection)
+    columns = list()
+    for col_spec in spec.columns:
+        key, col_type = col_spec.key, col_spec.type
+        if key not in docs[0]:
+            error_message = f'Unable to find column: {key} in the docs'
+            if allow_missing_metadata:
+                # some genome attribute tools may not have been executed for specific collection
+                print(error_message)
+                continue
+            else:
+                raise ValueError(error_message)
+
+        values = _convert_values_to_type(docs, key, col_type)
+        min_value, max_value, enum_values = None, None, None
+        if col_type in [ColumnType.INT, ColumnType.FLOAT, ColumnType.DATE]:
+            if values:
```

Comment:
What if all or all but one of the values are None?

Reply:
values returned by `_convert_values_to_type` will not have `None` value in it. If all entries for a column are empty, I updated to code to raise an error now. If values only have one value in it, both min and max value will be set to that value.

Reply:
I'm not sure I understand - it's seems possible that there could be, say, a numeric key where all the values across all genomes are None

Reply:
if all values across all genomes are None, the `values` variable returned by `_convert_values_to_type` will be an empty list `[]`. And an error will be raised when min/max_value is computed.

Reply:
Ok, I see how it works. The question then is is it reasonable for all the values for a key to be None? That seems possible to me

Reply:
👍 Added default min/max values when all values for a key is None.

Reply:
same comment re default values as above

Reply:
👍 set min/max to null if all values from a col are null.

Reply:
:+1: